CONTRIBUTING.md

```diff
@@ -1,61 +1,520 @@
+"SendObject", "UpdateObject", "StreamData". By default, you should not need to
+worry about AWS-service-specific API operations like `putItem()` with a unique
+hash -- or even which HTTP verbs to use.
+
+__Cascading service interactions__: Certain actions in a declarative style can have
+overlapping or ambiguous AWS Service implementations. With an opinionated
+implementation we can decide which Services are "primary" and which are
+"secondary" depending on what is configured. For instance sending an image will
+prefer S3 over API Gateway..
+
+__Simple, standard data objects__: Sending & Receiving data to AWS Services can
+have many parameters, which tend to show up in the SDKs. These are abstracted
+and inferred, where possible, with simple JSON that the implementation can
+reason about. Standard parameters (bucket names, stream names, partition keys,
+etc.) that are part of the implementation are extracted from a simplified
+configuration file and dynamically generated/updated in order to further allow
+focus on state and data types only.
+
+### Amplify Design
+As more and more modules are introduced to AWS Amplify, there is a clear
+necessity to modularize the library into smaller pieces so that users
+could avoid importing unnecessary parts into their app. The goal of this
+design is to make AWS Amplify modularized and also keep it backward
+compatible to avoid breaking changes.
+
+Modular import prevents unnecessary code dependencies are included with the
+app, and thus decreases the bundle size and enables adding new functionality
+without the risk of introducing errors related to the unused code.
+
+Amplify has established the concepts of categories and plugins. A
+__category__ is a collection of api calls that are exposed to the client
+to do things inside that category. For example, in the storage category
+generally one wants to upload and download objects from storage so the
+apis exposed to the client will represent that functionality. Because
+Amplify is pluggable, a __plugin__ of your choosing will provide the
+actual implementation behind that api interface.  Using the same example
+of Storage, the plugin we choose might be `AWSS3StoragePlugin`, which
+would then implement each api call from the category with a service call
+or set of service calls to S3, the underlying storage provider of the
+AWS plugin.
+
+## Getting Started
+
+First, ensure that you have installed the latest stable version of Android
+Studio / the Android SDK.
+
+Configure your environment, so that the `ANDROID_HOME` and `JAVA_HOME`
+environment variables are set. A convenient way of doing this is to add them
+into `~/.bashrc`. On a Mac, the SDK and Java installation used by the SDK may
+be found:
+
+```
+export ANDROID_HOME=~/Library/Android/sdk
+export JAVA_HOME=/Applications/Android\ Studio.app/Contents/jre/jdk/Contents/Home
+```
+Note: JDK 11, 12, 13, etc. have known issues and are not supported.
```

Comment:
This note is confusing. The latest Android Studio comes with the latest OpenJDK, right?

Reply:
Android Studio ships with JDK _1.8.0_212_:

```
/Applications/Android\ Studio.app/Contents/jre/jdk/Contents/Home/bin/java -version
openjdk version "1.8.0_212-release"
OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)
OpenJDK 64-Bit Server VM (build 25.212-b4-5784211, mixed mode)
```